packages/devtools_app/lib/src/flutter/auto_dispose_mixin.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+
+/// Provides functionality to automatically unsubscribe from streams on dispose.
+///
+/// Use this class to ensure you don't leak stream subscriptions
+mixin AutoDisposeMixin<T extends StatefulWidget> on State<T> {
+  final List<StreamSubscription> _subscriptions = [];
+
+  /// Track a stream subscription to be automatically cancelled on dispose.
+  void autoDispose(StreamSubscription subscription) {
+    _subscriptions.add(subscription);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
```

Comment:
good point! @pq this would be a nice lint so that other Flutter users who don't have an experienced Flutter user to review their code do the right thing.

Reply:
https://github.com/dart-lang/linter/issues/1780